zokrates_core/src/helpers/wasm.rs

```diff
@@ -0,0 +1,385 @@
+		let input = vec![FieldPrime::from(65536)];
+		let outputs = id.execute(&input);
+		assert_eq!(outputs, Err(String::from("Input #0 is stored on 3 bytes which is greater than the field size of 1")));
+
+		/* Code is the same as identity, tweaked so that field_size is a f32 */
+		let id = WasmHelper::new_code("0061736d010000000105016000017f030302000005030100010618047f0041010b7f0041010b7d0043000000000b7f0141000b074b06066d656d6f727902000e6765745f696e707574735f6f6666000105736f6c766500000a6d696e5f696e7075747303000b6d696e5f6f75747075747303010a6669656c645f73697a6503020a2f0228000340412023036a410023036a280200360200230341016a2403430000000023025c0d000b41200b040041000b0b4b020041000b20ffffffff000000000000000000000000ffffffff0000000000000000000000000041200b20deadbeef000000000000000000000000deadbeef000000000000000000000000");
+		let input = vec![FieldPrime::from(65536)];
+		let outputs = id.execute(&input);
+		assert_eq!(outputs, Err(String::from("Error converting `field_size` to i32")));
+	}
+
+	#[test]
+	fn check_identity() {
+		let id = WasmHelper::new_code(WasmHelper::IDENTITY_WASM);
+		let input = vec![FieldPrime::from(1)];
+		let outputs = id.execute(&input).expect("Identity call failed");
+		assert_eq!(outputs, input);
+
+		let id = WasmHelper::new_code(WasmHelper::IDENTITY_WASM);
+		let input = vec![FieldPrime::from(0)];
+		let outputs = id.execute(&input).expect("Identity call failed");
+		assert_eq!(outputs, input);
+	}
+
+	#[test]
+	fn check_identity_3_bytes() {
+		let id = WasmHelper::new_code(WasmHelper::IDENTITY_WASM);
+		let input = vec![FieldPrime::from(16777216)];
+		let outputs = id.execute(&input).expect("Identity call failed");
+		assert_eq!(outputs, input);
+	}
+
+	#[test]
+	fn check_invalid_arg_number() {
+		let id = WasmHelper::new_code(WasmHelper::IDENTITY_WASM);
+		let input = vec![FieldPrime::from(1)];
+		let outputs = id.execute(&input).expect("Identity call failed");
+		assert_eq!(outputs, input);
+	}
+
+	#[test]
+	fn check_memory_boundaries() {
+		/* Check that input writes are boundary-checked */
+		let id = WasmHelper::new_code("0061736d010000000105016000017f030302000005030100010615047f0041010b7f0041010b7f0041200b7f0141000b074b06066d656d6f727902000e6765745f696e707574735f6f6666000105736f6c766500000a6d696e5f696e7075747303000b6d696e5f6f75747075747303010a6669656c645f73697a6503020a2e0225000340412023036a410023036a280200360200230341016a240323032302470d000b41200b0600418080040b0b4b020041000b20ffffffff000000000000000000000000ffffffff0000000000000000000000000041200b20deadbeef000000000000000000000000deadbeef000000000000000000000000");
+		let input = vec![FieldPrime::from(65536)];
+		let outputs = id.execute(&input);
```

Comment:
I would expect this to pass, is this testing an issue with this particular module?

Reply:
It's about checking that if the module is buggy and returns an invalid offset, the correct error is reported.